server/jvm/src/test/scala/org/http4s/server/middleware/RouterSuite.scala

```diff
@@ -0,0 +1,27 @@
+package org.http4s.server.middleware
+
+import cats.effect._
+import org.http4s.Http4sSuite
+import org.http4s.HttpRoutes
+import org.http4s.Request
+import org.http4s.Status
+import org.http4s.server.Router
+import org.http4s.syntax.all._
+
+class RouterSuite extends Http4sSuite {
+  private val routes = {
+    import org.http4s.dsl.io._
+    HttpRoutes.of[IO] { case r @ GET -> Root / "foo" =>
+      assert(r.attributes.isEmpty, "Requests attributes should be empty")
```

Comment:
This isn't fancy but works. On the previous `Router#define` assert would fail. Also, if everybody would like this, I'll do a backport to 0.23.

Reply:
Thanks! Yes, running this up through 0.23 seems to be the way 👍 if you don't mind 😅

Reply:
Btw, why is it in JVM sources, does it not work on JS?

Reply:
mehhhh, I'll fix this

Reply:
Thanks for pointing that out, fixed.

Reply:
I was kind of surprised we didn't have a `RouterSuite` yet!